src/bp/core/services/dialog/flow/service.ts

```diff
@@ -87,13 +87,17 @@ export class FlowService {
 
   private _listenForCacheInvalidation() {
     this.cache.events.on('invalidation', async key => {
-      const matches = key.match(/^([A-Z0-9-_]+)::data\/bots\/([A-Z0-9-_]+)\/flows\/([\s\S]+(flow|ui)\.json)/i)
-
-      if (matches && matches.length >= 2) {
-        const [key, type, botId, flowName] = matches
-        if (type === 'file' || type === 'object') {
-          await this.forBot(botId).handleInvalidatedCache(flowName, type === 'file')
+      try {
+        const matches = key.match(/^([A-Z0-9-_]+)::data\/bots\/([A-Z0-9-_]+)\/flows\/([\s\S]+(flow|ui)\.json)/i)
+
+        if (matches && matches.length >= 2) {
+          const [key, type, botId, flowName] = matches
+          if (type === 'file' || type === 'object') {
+            await this.forBot(botId).handleInvalidatedCache(flowName, type === 'file')
+          }
         }
+      } catch (err) {
+        this.logger.error('Error Invalidating flow cache: ' + err.message)
```

Comment:
I am not sure I understand how this could happen? IMHO, if there is an error raised during the cache invalidation process, it's because there is a bug in the code the invalidation handler calls.

Reply:
I will give you an example that happened and the reason why this is here:
While pushing changes for BPFS, a flow for some reason had a "flow.json" but not an "ui.json", this resulted in a server crash and restart. In my option, this error should continue to occur but not crash the server.

Reply:
Okay, I see! The only thing that bugs me is the fact this scenario should not happen in the first place. But, I also understand that there are no real downsides in adding this exception handling!